app/src/main/java/com/github/braillesystems/learnbraille/data/repository/MaterialsRepository.kt

```diff
@@ -44,4 +49,7 @@ open class MaterialsRepositoryImpl(
         } else {
             deckDao.allDecks().map { DeckWithAvailability(it, true) }
         }
+
+    override suspend fun allDecksWithUniqueMaterials(): List<Deck> = DeckTags.Unique.values()
+        .mapNotNull { deckDao.deckByTag(it.tag) }
```

Comment:
A database query is executed for each Unique tag. It is significantly more efficient to grab all decks from the database and filter them.

Reply:
fixed this